shell/platform/common/client_wrapper/include/flutter/method_channel.h

```diff
@@ -13,9 +13,20 @@
 #include "method_call.h"
 #include "method_codec.h"
 #include "method_result.h"
+#include "standard_method_codec.h"
 
 namespace flutter {
 
+namespace {
+
+void ResizeChannel(BinaryMessenger* messenger, std::string name, int new_size);
+
+void SetChannelWarnsOnOverflow(BinaryMessenger* messenger,
+                               std::string name,
+                               bool warns);
+
+}  // namespace
+
```

Comment:
@stuartmorgan I really need your help here.
Using an anonymous namespace compile nicely locally and I can run unit test ( `.\out\host_debug_unopt\client_wrapper_unittests.exe`).

On CI, there is a compilation error related to the this forward reference.
The main issue is that those functions implementations call `MethodChannel`, so there is a kind of cyclic declaration issue. I thought using this forward declaration is the way to solved this, but it leads to the following error or CI:

```
../../flutter/shell/platform/common/client_wrapper/include/flutter/method_channel.h:167:6: error: function 'ResizeChannel' is not needed and will not be emitted [-Werror,-Wunneeded-internal-declaration]
  167 | void ResizeChannel(BinaryMessenger* messenger, std::string name, int new_size) {
      |      ^~~~~~~~~~~~~
../../flutter/shell/platform/common/client_wrapper/include/flutter/method_channel.h:185:6: error: function 'SetChannelWarnsOnOverflow' is not needed and will not be emitted [-Werror,-Wunneeded-internal-declaration]
  185 | void SetChannelWarnsOnOverflow(BinaryMessenger* messenger,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
```

If I remove the forward declaration, I still get no error locally but on CI there is a compilation error on lines 140 & 148 (the helpers functions can't be found, there are declared later (lines 167 & 185).

Reply:
> Using an anonymous namespace compiles nicely locally

Using an anonymous namespace in a header file is pretty much going to give good results.

If the intent is to signal that people should not call these methods directly, you can make a namespace called, e.g., `internal` that they are declared in, and comment that they should not be called from outside the file. There's no way to strictly enforce that with templated code, unfortunately.